src/planner/binder/expression/bind_macro_expression.cpp

```diff
@@ -9,43 +10,165 @@
 
 namespace duckdb {
 
-void ExpressionBinder::ReplaceMacroParametersRecursive(unique_ptr<ParsedExpression> &expr) {
+bool IsLambdaParameter(vector<unordered_set<string>> &lambda_params, ColumnRefExpression &col_ref) {
+	for (const auto &level : lambda_params) {
+		if (level.find(col_ref.GetColumnName()) != level.end()) {
+			return true;
+		}
+	}
+	return false;
+}
+
+void ExpressionBinder::ReplaceMacroParamInLambda(unique_ptr<ParsedExpression> &expr,
+                                                 vector<unordered_set<string>> &lambda_params) {
+
+	// special-handling for LHS lambda parameters
+	// we do not replace them, and we add them to the lambda_params vector
+	auto &lambda_expr = expr->Cast<LambdaExpression>();
+	lambda_params.emplace_back();
+
+	// single parameter, add to lambda_params
+	if (lambda_expr.lhs->expression_class == ExpressionClass::COLUMN_REF) {
+		auto column_ref = lambda_expr.lhs->Cast<ColumnRefExpression>();
+		lambda_params.back().emplace(column_ref.GetColumnName());
+
+	} else {
+		// multiple lambda parameters
+		auto &func_expr = lambda_expr.lhs->Cast<FunctionExpression>();
+		for (const auto &column_ref_expr : func_expr.children) {
+			auto column_ref = column_ref_expr->Cast<ColumnRefExpression>();
+			lambda_params.back().emplace(column_ref.GetColumnName());
+		}
+	}
+
+	// only replace in RHS
+	ParsedExpressionIterator::EnumerateChildren(
+	    *lambda_expr.expr, [&](unique_ptr<ParsedExpression> &child) { ReplaceMacroParameters(child, lambda_params); });
+
+	// pop this level
+	lambda_params.pop_back();
+}
+
+bool ExpressionBinder::MacroLambdaParamReplacer(unique_ptr<ParsedExpression> &expr,
+                                                vector<unordered_set<string>> &lambda_params) {
+
+	// early-out, if not a FUNCTION
+	if (expr->GetExpressionClass() != ExpressionClass::FUNCTION) {
+		return false;
+	}
+
+	auto &function = expr->Cast<FunctionExpression>();
+
+	// early-out for UNNEST
+	if (function.function_name == UNNEST_FUNCTION_ALIAS || function.function_name == UNLIST_FUNCTION_ALIAS) {
+		return false;
+	}
+
+	// early-out, if not in the catalog
+	QueryErrorContext error_context(binder.root_statement, function.query_location);
+	binder.BindSchemaOrCatalog(function.catalog, function.schema);
+	auto func = Catalog::GetEntry(context, CatalogType::SCALAR_FUNCTION_ENTRY, function.catalog, function.schema,
+	                              function.function_name, OnEntryNotFound::RETURN_NULL, error_context);
+	if (!func) {
+		return false;
+	}
+
+	// early-out, if not a SCALAR_FUNCTION_ENTRY
+	if (func->type != CatalogType::SCALAR_FUNCTION_ENTRY) {
+		return false;
+	}
+
+	// early-out, if JSON extension operator
+	if (function.function_name == "->>") {
```

Comment:
Do we need to do the function lookup before this early-out? Can't we do it in the same place where we do the early-out for unnest? Will this not be caught by the loop below that checks for `ExpressionClass::LAMBDA ` otherwise?